service/src/main/scala/za/co/absa/loginsvc/rest/config/provider/ConfigProvider.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.rest.config.provider
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import pureconfig._
+import pureconfig.generic.auto._
+import pureconfig.module.yaml._
+import za.co.absa.loginsvc.rest.config.actuator.GitConfig
+import za.co.absa.loginsvc.rest.config.auth._
+import za.co.absa.loginsvc.rest.config.validation.ConfigValidationException
+import za.co.absa.loginsvc.rest.config.{BaseConfig, JwtConfig}
+
+@Component
+class ConfigProvider(@Value("${spring.config.location}") yamlContent: String)
```

Comment:
This is a better than bofore. If you have the app launching in your hand directly, you can use this.

When you are e.g. building a Docker image and where want to set env variable, this becomes harder (btw. you cannot set env var containing `.`). I tried to change it to a different ENV var, e.g. `LS_CONFIG_LOCATION`), but that did not work either.

A viable option with this code seems to be update `JAVA_OPTS`, e.g.:
`ENV JAVA_OPTS="$JAVA_OPTS -Dspring.config.location=/opt/application.yaml"`

So it **seems to work** that way, but it is a bit harder to use.

What would help if there was a smarter config loader that would:
 - check the Java env var like now, e.g. `spring.config.location` or perhaps rather `ls.config.location`
 - check actual ENV VAR using `System.getenv("LS_CONFIG_LOCATION")`
- other options, possibly
With these option, there would be a priority ordering of which to use first.

I leave this comment up to you if you feel this should be implemented here or if you wish to create a future enhancement issue for this. As I said, there is a was to make it work for now using the `JAVA_OPTS` apparently from the system env var perspective.

Reply:
Ok, I'd rather just fix it now. I have an idea, let me just test it quick.

Reply:
Confirming that `SPRING_CONFIG_LOCATION` env var works for me 😄